beacon-chain/rpc/beacon_chain_server.go

```diff
@@ -319,7 +325,28 @@ func (bs *BeaconChainServer) ListValidatorBalances(
 			})
 		}
 	}
-	return &ethpb.ValidatorBalances{Balances: res}, nil
+
+	if len(res) == 0 {
```

Comment:
This the intended behavior? This reads, "if no balances matched the filter, return everything". 

So if I asked for a balance for an index that wasn't present in the state, i would get back all balances?

What about doing something like 
```go
if proto.Equal(req, &ethpb.GetValidatorBalancesRequest) {
   // return everything
}
```

hmm I guess you still need pagination though. In any case, I suggest to reconsider this approach.

Reply:
Great point, I added this logic to check if no indices or pubkeys were passed in, return everything, but forgot the case where nothing matched the filter. That approach seems best.